### PR TITLE
Add PostgreSQL cache

### PR DESCRIPTION
**Pull request by seyuboglu**
This cache works with GCP CloudSQL, making it easier to setup caches that are shared among users. 

Question: it introduces some optional dependencies 
```
   "pg8000",
        "google-cloud-storage",
        "sqlalchemy",
```
What's manifest's way of ensuring that these don't break imports for folks who don't have them installed?

### CODE REVIEW

**Comment by lorr1** [comment]
At the moment, a try/catch around the cache dictionary in `manifest.py`. So the postgres key only gets added if the class is loaded sucessfully.

File: manifest/manifest.py [hunk 1]
```diff
@@ -39,6 +40,7 @@
     "redis": RedisCache,
     "sqlite": SQLiteCache,
     "noop": NoopCache,
+    "postgres": PostgreSQLCache
```
Comment (lorr1) [inline manifest/manifest.py:43]: So try cache around here for import issues.

File: manifest/caches/postgres.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            cache_args: arguments for cache.
+        """
+        from google.cloud.sql.connector import Connector
+        connector = Connector()
+
+        def getconn():
+            conn = connector.connect(
+                connection_str,
+                "pg8000",
+                user=cache_args.pop("user"),
```
Comment (lorr1) [inline manifest/caches/postgres.py:49]: We probably want to set these as `pg_user` or `cache_user` or something so it's clear this is a cache param. Ideally we should refactor to get a clearer set of input params for a cache like we have for request, but this is fine for now. It'll happen once there are 2-3 caches that need params.

File: manifest/caches/postgres.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        engine = sqlalchemy.create_engine(
+            "postgresql+pg8000://",
+            creator=getconn,
+        )
+        engine.dialect.description_encoding = None
+
+        db_exists = len(sqlalchemy.inspect(engine).get_table_names()) > 0
+        if not db_exists:
+            print("Creating database...")
```
Comment (lorr1) [inline manifest/caches/postgres.py:63]: Logger.info?

**Comment by lorr1** [comment]
Also how hard would adding a test be. See https://github.com/HazyResearch/manifest/blob/c306e16da6d320f397db0705a278cd94158eeb44/tests/conftest.py#L20. I have support for instantiating different caches. If it's too hard or costs money, it's not worth it except maybe some get/set key methods. I certainly don't test all the clients lol.

**Review by lorr1 — CHANGES_REQUESTED** [review]
Fix tests formatting please!

**Comment by seyuboglu** [comment]
Just added in some simple tests that monkeypatch the connection to the db

File: manifest/caches/postgres.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
+"""SQLite cache."""
```
Comment (lorr1) [inline manifest/caches/postgres.py:1]: nit: Change comment to be postgres.

File: tests/conftest.py [hunk 5]
```diff
@@ -31,6 +31,17 @@ def redis_cache() -> Generator[str, None, None]:
         pass
 
 
+@pytest.fixture
+def postgres_cache(monkeypatch):
```
Comment (lorr1) [inline tests/conftest.py:35]: You probably need to add a type here. But this is pretty smart

**Review by lorr1 — APPROVED** [review]
A few nits and make sure tests pass.

**Pull request merged**